src/journal/journald-syslog.c

```diff
@@ -333,7 +336,9 @@ void server_process_syslog_message(
                 return;
 
         if (s->forward_to_syslog)
-                forward_syslog_raw(s, priority, orig, ucred, tv);
+                forward_syslog_raw(s, priority, (char *) orig, buf_len, ucred, tv);
+
+        buf = strstrip((char *) buf);
 
         syslog_skip_date((char**) &buf);
         syslog_parse_identifier(&buf, &identifier, &pid);
```

Comment:
hmm, so this code and what follows assumes the buffer is NUL terminated. Which i a bit weird, because you pass it in with a size value, which usually means "ignore NULs, the size is what matters". So now you have this hybrid logic where it's both size and NUL terminated and has to be. These are surprising calling conventions, unlike what is usually done...

This problem is solved with the snippet I proposed above, as it means we guarantee NUL termination